DopplerCustomDomain/Startup.cs

```diff
@@ -17,6 +18,9 @@ public Startup(IConfiguration configuration)
 
         public void ConfigureServices(IServiceCollection services)
         {
+            services.Configure<TraefikConfiguration>(Configuration.GetSection("TraefikConfiguration"));
+            services.AddScoped<ICustomDomainProviderService, CustomDomainProviderService>();
+            services.AddSingleton<IServiceNameResolver, ServiceNameResolver>();
```

Comment:
@andresmoschini here we have doubts about if is correct use `Scoped`, `Transient` or `Singleton` lifetimes

Reply:
In my opinion, singleton is fine because it does not depend on anything, it does not have state and it is thread-safe.